test/unit/io/sam_file/format_sam_test.cpp

```diff
@@ -125,49 +125,133 @@ struct sam_format : public sam_file_data
+        {
+            "@HD\tGO:none\tSO:coordinate\tVN:1.6\tSS:coordinate:queryname\n"
+            "@PG\tPN:novoalign\tPP:qc\tID:novoalign\tVN:V3.02.07\tCL:novoalign -d /hs37d5.ndx -f /file.fastq.gz\n"
+            "@SQ\tAS:hs37d5\tSN:ref2\tLN:243199373\n"
+            "@RG\tLB:1\tSM:NA12878\tPL:illumina\tPU:1\tID:U0a_A2_L1\n"
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_NO_THROW(fin.begin());
+    }
+    { // User defined tags should not trigger errors, but print warnings to cerr
+        std::string header_str
+        {
+            "@HD\tVN:1.6\tVB:user_tag\tSB:user_tag\tGB:user_tag\tpb:user_tag\n"
+            "@SQ\tSN:ref2\tLN:243199373\tSB:user_tag\tLB:user_tag\tpb:user_tag\n"
+            "@RG\tID:U0a_A2_L1\tIB:user_tag\tpb:user_tag\n"
+            "@PG\tID:qc\tIB:user_tag\tPB:user_tag\tCB:user_tag\tDB:user_tag\tVB:user_tag\tpb:user_tag\n"
+        };
+        std::string expected_cerr
+        {
+            "Unsupported SAM header tag in @HD: VB\n"
+            "Unsupported SAM header tag in @HD: SB\n"
+            "Unsupported SAM header tag in @HD: GB\n"
+            "Unsupported SAM header tag in @HD: pb\n"
+            "Unsupported SAM header tag in @PG: IB\n"
+            "Unsupported SAM header tag in @PG: PB\n"
+            "Unsupported SAM header tag in @PG: CB\n"
+            "Unsupported SAM header tag in @PG: DB\n"
+            "Unsupported SAM header tag in @PG: VB\n"
+            "Unsupported SAM header tag in @PG: pb\n"
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+
+        testing::internal::CaptureStderr();
+        EXPECT_NO_THROW(fin.begin());
+        EXPECT_EQ(testing::internal::GetCapturedStderr(), expected_cerr);
+    }
+    { // Required tags in HD
         std::string header_str
         {
-            "@HD\tVN:1.0\n"
-            "@PG\tID:prog\tTT:this is not a valid tag\n"
+            "@HD\n" // VN is missing
         };
         std::istringstream istream(header_str);
         seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
         EXPECT_THROW(fin.begin(), seqan3::format_error);
     }
+    // @SQ tests
+    {
+        std::string header_str
+        {
+            "@SQ\tLN:1\n"
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "The required SN tag in @SQ is missing."
+    }
     {
         std::string header_str
         {
-            "@HD\tVN:1.0\n"
-            "@SQ\tSN:unknown_ref\tLN:0\n"
+            "@SQ\tSN:ref\n"
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "The required LN tag in @SQ is missing or 0."
```

Comment:
```suggestion
        EXPECT_THROW(fin.begin(), seqan3::format_error);
```

Also, this would raise an error that LN is missing since we handle that case explicitly